std/math.d

```diff
@@ -1382,10 +1390,18 @@ real cosh(real x) @safe pure nothrow @nogc
 }
 
 /// ditto
-double cosh(double x) @safe pure nothrow @nogc { return cosh(cast(real) x); }
+double cosh(double x) @safe pure nothrow @nogc
+{
+    const double y = exp(x);
+    return (y + 1.0/y) * 0.5;
+}
 
 /// ditto
-float cosh(float x) @safe pure nothrow @nogc  { return cosh(cast(real) x); }
+float cosh(float x) @safe pure nothrow @nogc
+{
+    const float y = exp(x);
+    return (y + 1.0/y) * 0.5;
```

Comment:
Use the `f` suffix for the literals, otherwise it'll be computed in double precision before being truncated to float.